pkg/orchestration/wiring/edge/plugin.go

```diff
@@ -0,0 +1,92 @@
+package edge
+
+import (
+	"encoding/json"
+	smith "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager"
+	orchestration "github.com/atlassian/voyager/pkg/apis/orchestration/v1"
+	wiring "github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/osb"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	ResourceType voyager.ResourceType = "GlobalEdge"
+	ServiceId                         = "10e5a402-45df-5afd-ae86-11377ce2bbb2"
+	PlanId                            = "7d57270a-0348-58d3-829d-447a98fe98d5"
+)
+
+type WiringPlugin struct {
+}
+
+func New() *WiringPlugin {
+	return &WiringPlugin{}
+}
+
+func (p *WiringPlugin) WireUp(resource *orchestration.StateResource, context *wiring.WiringContext) (result *wiring.WiringResult, retriable bool, err error) {
```

Comment:
Please don't use named return values and naked returns. They are a source of bugs. Also for codebase style consistency.

Reply:
Removed naked returns from this